tinnsleep/check_impedance.py

```diff
@@ -0,0 +1,156 @@
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_electrodes)
+        Root Mean Square Amplitude
+        check_imp list of list of 0s and 1s, shape (nb_epochs, nb_electrodes) 0s marking bad channels for the designated
+        epoch
+
+        Returns
+        -------
+        X : ndarray, shape (n_trials, n_electrodes)
+        Modified Root Mean Square Amplitude
+        """
+    # Verify that X and check_imp have the same shape
+    # TO DO Throw an error if else
+    if not (len(X) == len(check_imp) and len(X[0]) == len(check_imp[0])) :
+        raise ValueError("Inputs shapes don't match")
+
+    mod_X=[]
+    for i in range(len(X)):
+        if not np.mean(check_imp[i]) == 1:
+            remaining_chan = []
+            bad_chan_ind = []
+            mod_X.append(X[i])
+            for j in range(len(X[0])):
+                if check_imp[i][j] == 0:
+                    remaining_chan.append(X[i][j])
+                else:
+                    bad_chan_ind.append(j)
+            rem_chan_av = np.mean(remaining_chan)
+            for ind in bad_chan_ind:
+                mod_X[-1][ind] = rem_chan_av
+
+    return mod_X
+
+def fuse_with_classif_result(check_imp, labels):
+    """Adds at the good indexes the missing elements of labels because of the use of check_RMS
+            Parameters
+            ----------
+            check_imp list of list of 0s and 1s, shape (nb_epochs, nb_electrodes) 0s marking bad channels for the
+            designated epoch
+            labels : ndarray, shape (n_trials - nb_bad_epochs,) array of the classification prediction results
+
+            Returns
+            -------
+            mod_labels : ndarray, shape (n_trials,)
+            Modified labels to fit a length of n_trials
+            """
+    labels = labels.tolist()
+    for i in range(len(check_imp)):
+        if np.mean(check_imp[i]) == 1:
+            labels.insert(i, False)
+    return labels
+
+
+def create_bad_epochs_annotation(check_imp, duration, interval, orig_time=0.0):
+    """Create a list of annotations of all the bad epochs where all channels have abnormal impedance values to annotate
+    the raw file
+           Parameters
+           ----------
+           check_imp list of list of 0s and 1s, shape (nb_epochs, nb_electrodes) 0s marking bad channels for the designated
+           epoch
+           duration: int
+           Number of elements (i.e. samples) on the epoch.
+           interval: int
+           Number of elements (i.e. samples) to move for the next epoch.
+
+           Returns
+           -------
+           annotations : list of dictionaries
+           Chronological list of bad epoch annotations
+           """
+    annotations = []
+    for i in range(len(check_imp)):
+        if np.mean(check_imp[i]) == 1:
+            annotations.append({'onset': i * interval, 'duration': duration, 'description': "1", 'orig_time': orig_time})
+    return annotations
```

Comment:
As the scheme is with fixed interval, I advice using the `_sliding` keyword in such fun. Not important for now but nicer for consistency.